### PR TITLE
ocamlPackages.ocamlformat: Move into 'ocamlPackages'

### PR DESCRIPTION
**Pull request by Julow**
###### Description of changes

Move the definitions of OCamlformat packages into ocamlPackages. This moves the package containing the tool closer to the definition of the library.

It's now possible to build OCamlformat using a different version of OCaml than the default.

`ocamlformat`, `ocamlPackages.ocamlformat` and
`ocaml-ng.ocamlPackages_<lastest>.ocamlformat` are all aliases.

This follows many other ocaml tools (`dune`, `patdiff`, `ocaml-crunch`, `reason`, `wasm`, and others), though many ocaml tools are still defined outside of `ocamlPackages` (29 in `pkgs/development/tools/ocaml/`).
This was suggested by @anmonteiro and @Zimmi48 in https://github.com/NixOS/nixpkgs/pull/227229

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Alizter — APPROVED** [review]

File: pkgs/development/ocaml-modules/ocamlformat/ocamlformat.nix [hunk 1]
```diff
@@ -2,11 +2,7 @@
 
 with ocaml-ng.ocamlPackages;
```
Comment (anmonteiro) [inline pkgs/development/ocaml-modules/ocamlformat/ocamlformat.nix:3]: this shouldn't open `ocaml-ng.ocamlPackages`. It's still using the default OCaml version. 

Instead, the library dependencies should be passed as arguments.

**Review by anmonteiro — COMMENTED** [review]
you also need to change ocamlformat/generic.nix to not take `ocaml-ng` as an argument

File: pkgs/development/ocaml-modules/ocamlformat/ocamlformat.nix (hunk 1 above)
Comment (Julow) [reply pkgs/development/ocaml-modules/ocamlformat/ocamlformat.nix]: Good catch thanks!

This bug was here before and made the library incompatible when using a non-default version. Fixed!

**Comment by Alizter** [comment]
Result of `nixpkgs-review pr 241708` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>ocamlPackages.ocamlformat (ocamlPackages.ocamlformat_0_25_1)</li>
    <li>ocamlPackages.ocamlformat_0_19_0</li>
    <li>ocamlPackages.ocamlformat_0_20_0</li>
    <li>ocamlPackages.ocamlformat_0_20_1</li>
    <li>ocamlPackages.ocamlformat_0_21_0</li>
    <li>ocamlPackages.ocamlformat_0_22_4</li>
    <li>ocamlPackages.ocamlformat_0_23_0</li>
    <li>ocamlPackages.ocamlformat_0_24_1</li>
  </ul>
</details>

**Review by anmonteiro — APPROVED** [review]
thank you, @julow!

**Review by ulrikstrid — APPROVED** [review]
Looks good 👍

**Comment by anmonteiro** [comment]
> though many ocaml tools are still defined outside of `ocamlPackages` (29 in `pkgs/development/tools/ocaml/`).

Only 8 of these are not present in ocamlPackages:

```
obelisk
obuild
ocaml-top
ocsigen-i18n
omake
opaline
opam 
opam2json
```

**Pull request merged**